spacelift/data_worker_pools.go

```diff
@@ -0,0 +1,94 @@
+package spacelift
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/structs"
+)
+
+func dataWorkerPools() *schema.Resource {
+	return &schema.Resource{
+		Read: dataWorkerPoolsRead,
+		Schema: map[string]*schema.Schema{
+			"worker_pools": &schema.Schema{
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"worker_pool_id": {
+							Type:        schema.TypeString,
+							Description: "ID of the worker pool",
+							Required:    true,
```

Comment:
If it's a data source, then parameters which are read and don't have to be provided, should be marked as Computer: true, instead of Required: true